Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/ApiGatewayHttpApiV2ProxyRequestTranslator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Amazon.Lambda.APIGatewayEvents;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using static Amazon.Lambda.APIGatewayEvents.APIGatewayHttpApiV2ProxyRequest;
+
+namespace Amazon.Lambda.TestTool
+{
+    /// <summary>
+    /// Translates ASP.NET Core HTTP requests to API Gateway HTTP API v2 Proxy Requests.
+    /// </summary>
+    public class ApiGatewayHttpApiV2ProxyRequestTranslator : IApiGatewayRequestTranslator
+    {
+        private readonly IHttpRequestUtility _httpRequestUtility;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiGatewayHttpApiV2ProxyRequestTranslator"/> class.
+        /// </summary>
+        /// <param name="httpRequestUtility">The HTTP request utility used for extracting request details.</param>
+        public ApiGatewayHttpApiV2ProxyRequestTranslator(IHttpRequestUtility httpRequestUtility)
+        {
+            _httpRequestUtility = httpRequestUtility;
+        }
+
+        /// <summary>
+        /// Translates an ASP.NET Core HTTP request to an API Gateway HTTP API v2 Proxy Request.
+        /// </summary>
+        /// <param name="request">The ASP.NET Core HTTP request to translate.</param>
+        /// <param name="pathParameters">The path parameters extracted from the request URL.</param>
+        /// <param name="resource">The API Gateway resource path.</param>
+        /// <returns>An <see cref="APIGatewayHttpApiV2ProxyRequest"/> object representing the translated request.</returns>
+        public object TranslateFromHttpRequest(HttpRequest request, IDictionary<string, string> pathParameters, string resource)
+        {
+            var (headers, _) = _httpRequestUtility.ExtractHeaders(request.Headers);
+            var (queryStringParameters, _) = _httpRequestUtility.ExtractQueryStringParameters(request.Query);
+
+            var httpApiV2ProxyRequest = new APIGatewayHttpApiV2ProxyRequest
+            {
+                RouteKey = $"{request.Method} {resource}",
+                RawPath = request.Path,
+                RawQueryString = request.QueryString.Value,
+                Cookies = request.Cookies.Select(c => $"{c.Key}={c.Value}").ToArray(),
+                Headers = headers,
+                QueryStringParameters = queryStringParameters,
+                PathParameters = pathParameters ?? new Dictionary<string, string>(),
```

Comment:
Could you please extract the path parameters within the translation layer?

Reply:
for the path parameter extraction, i think i originally had it inside the translate function. but because i would need to get the matching config too and stuff i took it out and opted for an input parameter.

i will update the code to use extensions to see how that looks first

Reply:
https://github.com/aws/aws-lambda-dotnet/pull/1902#discussion_r1878517289 from this comment, i will need route config in order to extract the path parameters. so either ill need to re-get it, or just have another abstraction on top of the translation layer which takes in the route config. we can discuss more